core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -79,7 +79,6 @@
 import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
 import static io.trino.sql.tree.ComparisonExpression.Operator.GREATER_THAN;
 import static io.trino.sql.tree.ComparisonExpression.Operator.GREATER_THAN_OR_EQUAL;
-import static io.trino.sql.tree.ComparisonExpression.Operator.IS_DISTINCT_FROM;
```

Comment:
error-prone's RemoveUnusedImports doesn't seem to catch this either

cc @ksobolew

Reply:
I reported a similar bug in Checkstyle: checkstyle/checkstyle#12923. It concerns statically-imported methods, but maybe it also misses constants as well when they are used as targets in `switch`.

Reply:
it looks like error-prone would be in a better position to correctly analyze unused imports, because it also understands class inheritance. would you want to try to report this problem with RemoveUnusedImports to error-prone?

Reply:
First, I reported it to Checkstyle: checkstyle/checkstyle#13361, because there is a very similar issue reported there already (checkstyle/checkstyle#12923).

Reply:
And here's a mostly copy-pasted one in Error Prone: google/error-prone#4000 (Error Prone 4000!)

Reply:
And I have a weird feeling of deja vu, as if I already reported something like this somewhere, but I can't find any previous issues...

Reply:
thank you @ksobolew !

Reply:
FYI: The checkstyle issue was closed because Checkstyle does not have the necessary type information to reliably detect this. This leaves us with ErrorProne (but no movement on that front).